libraries/rush-lib/src/cli/test/RushCommandLineParser.test.ts

```diff
@@ -317,5 +317,172 @@ describe('RushCommandLineParser', () => {
+          expect(packageCount).toEqual(2);
+
+          // Use regex for task name in case spaces were prepended or appended to spawned command
+          const expectedBuildTaskRegexp: RegExp = /fake_build_task_but_works_with_mock/;
+
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const firstSpawn: any[] = spawnMock.mock.calls[0];
+          expect(firstSpawn[SPAWN_ARG_ARGS]).toEqual(
+            expect.arrayContaining([expect.stringMatching(expectedBuildTaskRegexp)])
+          );
+          expect(firstSpawn[SPAWN_ARG_OPTIONS]).toEqual(expect.any(Object));
+          pathEquals(firstSpawn[SPAWN_ARG_OPTIONS].cwd, `${repoPath}/a`);
+
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const secondSpawn: any[] = spawnMock.mock.calls[1];
+          expect(secondSpawn[SPAWN_ARG_ARGS]).toEqual(
+            expect.arrayContaining([expect.stringMatching(expectedBuildTaskRegexp)])
+          );
+          expect(secondSpawn[SPAWN_ARG_OPTIONS]).toEqual(expect.any(Object));
+          pathEquals(secondSpawn[SPAWN_ARG_OPTIONS].cwd, `${repoPath}/b`);
+        });
+      });
+
+      describe("'rebuild' action", () => {
+        it(`executes the package's 'rebuild' script`, async () => {
+          const repoName: string = 'pluginWithRebuildCommandRepo';
+          const { parser, spawnMock, repoPath } = await getCommandLineParserInstanceAsync(
+            repoName,
+            'rebuild'
+          );
+
+          await expect(parser.executeAsync()).resolves.toEqual(true);
+
+          // There should be 1 build per package
+          const packageCount: number = spawnMock.mock.calls.length;
+          expect(packageCount).toEqual(2);
+
+          // Use regex for task name in case spaces were prepended or appended to spawned command
+          const expectedBuildTaskRegexp: RegExp = /fake_REbuild_task_but_works_with_mock/;
+
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const firstSpawn: any[] = spawnMock.mock.calls[0];
+          expect(firstSpawn[SPAWN_ARG_ARGS]).toEqual(
+            expect.arrayContaining([expect.stringMatching(expectedBuildTaskRegexp)])
+          );
+          expect(firstSpawn[SPAWN_ARG_OPTIONS]).toEqual(expect.any(Object));
+          pathEquals(firstSpawn[SPAWN_ARG_OPTIONS].cwd, `${repoPath}/a`);
+
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const secondSpawn: any[] = spawnMock.mock.calls[1];
+          expect(secondSpawn[SPAWN_ARG_ARGS]).toEqual(
+            expect.arrayContaining([expect.stringMatching(expectedBuildTaskRegexp)])
+          );
+          expect(secondSpawn[SPAWN_ARG_OPTIONS]).toEqual(expect.any(Object));
+          pathEquals(secondSpawn[SPAWN_ARG_OPTIONS].cwd, `${repoPath}/b`);
+        });
+      });
+    });
+
+    describe('in repo plugin with conflict build command', () => {
+      it(`throws an error when starting Rush`, async () => {
```

Comment:
Seems like failing cases are being put in **RushCommandLineParserFailureCases.test.ts** instead

Reply:
I do have this doubt, but seeing that there are also incorrect test cases in #257, #269, and #281, I thought I could put it in this file.

If this is indeed how it is designed, I can move it over.

Reply:
Let's not worry about it for now then.